File: src/levels/L06-decorator/ts/adapter.ts
class Menu {
  // region: Menu#drink_for
  drinkFor(request: DrinkRequest): Drink {
    return new LegacyMenuAdapter(new OldPos()).drinkFor(request)
  }
  // endregion
}

// Traduce el pedido al formato del punto de venta viejo… que tampoco conoce combinaciones.
// region: LegacyMenuAdapter
class LegacyMenuAdapter {
  constructor(private pos: OldPos) {}
  drinkFor(request: DrinkRequest): Drink { return this.pos.lookup(request.extras.join('+')) }
}
// endregion

File: src/levels/L06-decorator/adapter.rb
class Menu
  # region: Menu#drink_for
  def drink_for(request)
    LegacyMenuAdapter.new(OldPos.new).drink_for(request)
  end
  # endregion
end

# Traduce el pedido al formato del punto de venta viejo… que tampoco conoce combinaciones.
# region: LegacyMenuAdapter
class LegacyMenuAdapter
  def initialize(pos) = @pos = pos
  def drink_for(request) = @pos.lookup(request.extras.join("+"))
end
# endregion
